xds/src/main/java/io/grpc/xds/XdsLoadBalancer.java

```diff
@@ -360,6 +340,38 @@ void updateFallbackServers(
       }
     }
 
+    private void handleFallbackAddressesOrNameResolutionError() {
+      String fallbackPolicyName = fallbackPolicy.getPolicyName();
+      List<EquivalentAddressGroup> servers = fallbackServers;
+
+      // Some addresses in the list may be grpclb-v1 balancer addresses, so if the fallback policy
+      // does not support grpclb-v1 balancer addresses, then we need to exclude them from the list.
+      if (!fallbackPolicyName.equals("grpclb") && !fallbackPolicyName.equals(XDS_POLICY_NAME)) {
+        ImmutableList.Builder<EquivalentAddressGroup> backends = ImmutableList.builder();
+        for (EquivalentAddressGroup eag : fallbackServers) {
+          if (eag.getAttributes().get(GrpcAttributes.ATTR_LB_ADDR_AUTHORITY) == null) {
+            backends.add(eag);
+          }
+        }
+        servers = backends.build();
+      }
+
+      // FIXME: this is a temporary hack.
```

Comment:
Please add TODO(zhangkun83) and refer to https://github.com/grpc/grpc-java/issues/5496

Reply:
Done.